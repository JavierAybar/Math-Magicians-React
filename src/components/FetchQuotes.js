import { useEffect, useState } from 'react';
import { v4 as uuidv4 } from 'uuid';

const FetchQuotes = () => {
  const [data, setData] = useState([]);
  const [error, setError] = useState(null);
  const [isLoading, setIsLoading] = useState(false);

  useEffect(() => {
    const fetchData = async () => {
      setIsLoading(true);
      try {
        const url = 'https://api.api-ninjas.com/v1/quotes?category=alone';
        const response = await fetch(url, {
          headers: {
            'X-Api-Key': 'pB33PTqHltghkmhwcxN9dw==XpkYpcTp7PCuAB1v',
          },
        });
        const json = await response.json();
        setData(json);
      } catch (error) {
        setError(error);
      }
      setIsLoading(false);
    };
    fetchData();
  }, []);

  if (error) return <div className="error">Something went wrong!</div>;

  if (isLoading) return <div className="loading">Loading...</div>;

  return (
    <div className="container__quotes">
      <h2 className="title__quotes">Quotes</h2>
      <ul className="quotes">
        {data.map((item, index) => (/* eslint-disable-next-line react/no-array-index-key */
          <div className="section-quotes" key={uuidv4() + index}>
            <li>{item.quote}</li>
            <li>
              <strong>Author:</strong>
              {' '}
              {item.author}
            </li>
          </div>
        ))}
      </ul>
    </div>
  );
};

export default FetchQuotes;
