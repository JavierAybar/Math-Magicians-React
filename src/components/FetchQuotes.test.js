import { render, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import FetchQuotes from './FetchQuotes';

describe('FetchQuotes component', () => {
  test('renders loading message', () => {
    const { container } = render(<FetchQuotes />);
    expect(container).toMatchSnapshot();
  });

  test('renders error message when fetch fails', async () => {
    jest.spyOn(global, 'fetch').mockRejectedValue(new Error('Network Error'));

    await (async () => {
      const { container } = render(<FetchQuotes />);
      await waitFor(() => expect(container).toMatchSnapshot());
    });
  });

  test('renders quotes after successful fetch', async () => {
    jest.spyOn(global, 'fetch').mockResolvedValue({
      json: async () => [{ quote: 'Quote 1', author: 'Author 1' }],
    });

    await (async () => {
      const { container } = render(<FetchQuotes />);
      await waitFor(() => expect(container).toMatchSnapshot());
    });
  });
});
