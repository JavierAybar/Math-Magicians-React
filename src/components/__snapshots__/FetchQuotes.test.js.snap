// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FetchQuotes component renders loading message 1`] = `
<div>
  <div
    class="loading"
  >
    Loading...
  </div>
</div>
`;
